src/mbed_tools/cli/build.py

```diff
@@ -94,13 +95,17 @@ def build(
     build_tree = program.files.cmake_build_dir
     if clean and build_tree.exists():
         shutil.rmtree(build_tree)
-
-    if any([not mbed_config_file.exists(), not build_tree.exists(), mbed_target, toolchain]):
+    if any(
+        [
+            not mbed_config_file.exists(),
+            not build_tree.exists(),
+            mbed_os_path is not None,
+            custom_targets_json is not None,
+        ]
```

Comment:
I had thought if the user has a custom_target_json arg then they would want to configure, as that won't be used otherwise, but maybe it would be better if they had to explicitly choose to reconfigure.

Reply:
Why don't we just always reconfigure? CMake should be smart enough to figure out if we need an incremental build or not.